### PR TITLE
HW-DB

### PR DESCRIPTION
**Pull request by Max-Pochepets**
DATABASE connection established

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -18,6 +18,14 @@
         </maven.checkstyle.plugin.configLocation>
     </properties>
```
Comment (vicire) [inline pom.xml]: remove empty line here and under

File: src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+public class ManufacturerDaoJdBcImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String query = "INSERT INTO manufacturers (name,country) VALUES (?,?)";
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement
+                    = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            setUpdate(preparedStatement, manufacturer).executeUpdate();
+
```
Comment (vicire) [inline src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java]: remove empty lines :)

File: src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                manufacturer.setId(resultSet.getObject(1, Long.class));
+            }
+            preparedStatement.close();
+
+            return manufacturer;
+        } catch (SQLException throwables) {
+            throw new RuntimeException("Couldn't establish connection to MySQL server. "
```
Comment (vicire) [inline src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java]: look in common mistakes. You need to create the custom exception.

File: src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String query = "SELECT id, name, country FROM manufacturers WHERE id = "
+                + id + " AND deleted = FALSE";
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            Statement statement = connection.createStatement();
+            ResultSet resultSet = statement.executeQuery(query);
+            Manufacturer manufacturer = new Manufacturer("", "");
+            while (resultSet.next()) {
```
Comment (vicire) [inline src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java]: Same here - you can create a private method for logic in cycle

File: src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+        String query = "UPDATE manufacturers SET name = ?, country = ?"
+                + " WHERE id = ? AND deleted = FALSE";
+        String getNewQuery = "SELECT * FROM manufacturers WHERE id = " + manufacturer.getId();
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement
+                    = setUpdate(connection.prepareStatement(query), manufacturer);
+            preparedStatement.setLong(3, manufacturer.getId());
+            preparedStatement.executeUpdate();
+            preparedStatement.close();
```
Comment (vicire) [inline src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java]: Honestly, that`s enough for this method) you can return manufacturer, that you already have

File: src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE manufacturers SET deleted = TRUE WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
```
Comment (vicire) [inline src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java:94]: Create constants for magic

**Review by vicire — CHANGES_REQUESTED** [review]
Nice)

File: src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java (hunk 6 above)
Comment (Max-Pochepets) [reply src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java:94]: I can't, cause this indexes don't serve the same purpose every time they are used.

**Review by vicire — APPROVED** [review]

File: src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+import java.util.Optional;
+
+@DaoImpl
+public class ManufacturerDaoJdBcImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String query = "INSERT INTO manufacturers (name,country) VALUES (?,?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement
+                    = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
```
Comment (Sofasmile) [inline src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java]:
```suggestion
        try (Connection connection = ConnectionUtil.getConnection();
            PreparedStatement preparedStatement
                    = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
```
and then we can remove `statement.close()`

File: src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+            setUpdate(preparedStatement, manufacturer).executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                manufacturer.setId(resultSet.getObject(1, Long.class));
+            }
+            preparedStatement.close();
+            return manufacturer;
+        } catch (SQLException throwables) {
+            throw new DataBaseException("Couldn't create manufacturer.",
```
Comment (Sofasmile) [inline src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java]: which manufacturer can not be created?

File: src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+            throw new DataBaseException("Couldn't get a list of manufacturers.",
+                    throwables);
+        }
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String query = "UPDATE manufacturers SET name = ?, country = ?"
+                + " WHERE id = ? AND deleted = FALSE";
+        String getNewQuery = "SELECT * FROM manufacturers WHERE id = " + manufacturer.getId();
```
Comment (Sofasmile) [inline src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java]: do we need this query here?

File: src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+            Statement statement = connection.createStatement();
+            ResultSet resultSet = statement.executeQuery(query);
+            Manufacturer manufacturer = null;
+            if (resultSet.next()) {
+                manufacturer = setManufacturer(resultSet);
+            }
+            statement.close();
+            return Optional.ofNullable(manufacturer);
+        } catch (SQLException throwables) {
+            throw new DataBaseException("Couldn't get manufacturer by id.",
```
Comment (Sofasmile) [inline src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java]: let's make all messages more informative

File: src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException throwables) {
+            throw new DataBaseException("Couldn't delete a manufacturer by id.",
+                    throwables);
+        }
+    }
+
+    private static Manufacturer setManufacturer(ResultSet resultSet) throws SQLException {
```
Comment (Sofasmile) [inline src/main/java/crud/dao/impl/ManufacturerDaoJdBcImpl.java]: why is this method static?

**Review by Sofasmile — CHANGES_REQUESTED** [review]

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**